src/Tribe/Repository/Read_Interface.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+/**
+ * Interface Tribe__Repository__Read_Interface
+ *
+ * @since TBD
+ */
+interface Tribe__Repository__Read_Interface {
+	const PERMISSION_EDITABLE = 'editable';
+	const PERMISSION_READABLE = 'readable';
+
+	/**
+	 * Batch filter application method.
+	 *
+	 * This is the same as calling `by` multiple times with different arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $args An associative array of arguments to filter
+	 *                    the posts by in the shape [ <key>, <value> ].
+	 *
+	 * @return $this|Tribe__Repository__Read_Interface
+	 */
+	public function by_args( array $args );
+
+	/**
+	 * Applies a filter to the query.
+	 *
+	 * While the signature only shows 2 arguments additional arguments will be passed
+	 * to the schema filters.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key
+	 * @param mixed  $value
+	 * @param mixed  ...$args Additional, optional, call arguments that will be passed to
+	 *                        the schema.
+	 *
+	 * @return $this|Tribe__Repository__Read_Interface
+	 */
+	public function by( $key, $value );
+
+	/**
+	 * Just an alias of the `by` method to allow for easier reading.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key
+	 * @param mixed  $value
+	 *
+	 * @return Tribe__Repository__Read|Tribe__Repository__Read_Interface
+	 */
+	public function where( $key, $value );
+
+	/**
+	 * Sets the page of posts to fetch.
+	 *
+	 * Mind that this implementation does not support a `by( 'page', 2 )`
+	 * filter to force more readable code.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $page
+	 *
+	 * @return $this|Tribe__Repository__Read_Interface
+	 */
+	public function page( $page );
+
+	/**
+	 * Sets the number of posts to retrieve per page.
+	 *
+	 * Mind that this implementation does not support a `by( 'per_page', 5 )`
+	 * filter to force more readable code; by default posts per page is `-1`.
+	 *
+	 * @param int $per_page
+	 *
+	 * @return $this|Tribe__Repository__Read_Interface
+	 */
+	public function per_page( $per_page );
+
+	/**
+	 * Returns the number of posts found matching the query.
+	 *
+	 * @return int
+	 */
+	public function found();
```

Comment:
I wonder if this should be `count()` rather than `found()`. The former seems more indicative of the return type and the latter seems more like it would return a boolean.

Now that I type that, there's probably some english-language uses where `found()` makes more sense. Maybe supporting both would have its place (alias `count()` for `found()`)?

Reply:
The idea behind `found` is to stick with the `$wp_query->found_posts` convention.
But aliasing it to `count` makes a lot of sense too.

Reply:
If this is meant to be found() => $wp_query->found_posts, this makes a lot of sense. For Pods we use total() and total_found() which do two different things.

* total() returns the total number of results in the current result set
* total_found() returns the total number of results across all pages of the result query

Reply:
My vote is to not alias count() with found(), but to return a count of the $wp_query->posts in count()

Reply:
This is a neat idea!
Let me update the code and try it out!